### ISSUE TITLE
Add option to change leader-board decimal precision

### ISSUE THREAD

**Issue by Sanji515** [Cloud-CV/EvalAI-ngx#284]
Add option under settings tab in challenge page to change leader-board decimal precision by challenge host (**Note:** settings tab is only for the host user).

It is expected to add the front-end to be able to show the current `leaderbord decimal precision` value (refer field [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/models.py#L328)) and to add the option to change this field by the host user.

**Comment by Suryansh5545** [comment]
This was a GCI task and the problem faced by the student was that they were able to find how precision value is changed or defined so. For those who will attempt this you can find the precision value here https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html#L114 this value is currently statically applied.

**Comment by Abhi58** [comment]
Hi @Sanji515 @Suryansh5545 . I would like to take up this issue. Please assign this to me. Thanks!

**Comment by Sanji515** [comment]
Sure @Abhi58 You can work on it.

**Comment by Abhi58** [comment]
Hi @Sanji515 . I was going through the code of EvalAI. There i found a variable called 'leaderboard_decimal_precision' which i believe is a part of callback response data on line 1437 in controller.js. I searched for the variable in the controller file, but couldn't find it.

**Comment by Abhi58** [comment]
Could you tell me from where is the response coming from? Thanks!

**Comment by Sanji515** [comment]
@Abhi58 I didn't get exactly of which controller.js file you are pointing line 1437. But you can take a look at this API [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/views.py#L1748)

**Comment by Abhi58** [comment]
Sure @Sanji515 . Thanks!

**Comment by Abhi58** [comment]
Hi @Sanji515 . I ran the EvalAI-ngx using ng serve because docker-compose up was giving errors. So, based on EvalAI code and the assumption that the backend is same for EvalAI and ngx i have made some changes.

**Comment by Abhi58** [comment]
I am not able to test my code since the backend server is not running. I will create a work in progress PR. Please verify it and let me know if the issue has been resolved. Sorry for the trouble! Thanks!

**Comment by Suryansh5545** [comment]
@abhi58 What problem are you facing?

**Comment by Suryansh5545** [comment]
@Abhi58 I think you can edit the precision value here https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html#L114 also if your docker setup can't run angular container then you can try to keep both backend and frontend started on two terminals.

**Comment by Abhi58** [comment]
Hi @Suryansh5545 . Backend was not running. That's the problem i was facing. As you suggested i will try to run on two terminals.

**Comment by Abhi58** [comment]
Actually, i have made changes to the line that you have mentioned based on the original EvalAI code. Please review it and let me know. Thanks! @Suryansh5545

**Comment by Suryansh5545** [comment]
@Abhi58 i have made some comments please take a look

**Comment by Kajol-Kumari** [comment]
@Abhi58  May i know if you are still working on it or not as i would like to work on it.

**Comment by Abhi58** [comment]
@Kajol-Kumari yes, i am still working on it. I was busy with school work this week so couldn't spend much time. I will be getting back to working on this today. Please pick another issue. Thanks!

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari yes, i am still working on it. I was busy with school work this week so couldn't spend much time. I will be getting back to working on this today. Please pick another issue. Thanks!

@Abhi58 I have already done some of its part, so if you wish,i can create a draft PR which you can pull and continue to work.If you want,Please let me know.

**Comment by Abhi58** [comment]
@Kajol-Kumari That's okay. I have finished working on it. I will create a pull request soon. You don't have to create a pr. Actually, if you see i already created a pull request before, but that had some problems. So, i just had to do some small changes. Sorry about that! Please feel free to pick another issue.

**Comment by Abhi58** [comment]
@Sanji515 @Suryansh5545 @RishabhJain2018 I have raised a pr #290 which resolves this issue. Please review it and close this issue.
